spyder/plugins/help/widgets.py

```diff
@@ -473,46 +465,66 @@ def setup(self, options):
         self.plain_text.sig_custom_context_menu_requested.connect(
             self._show_plain_text_context_menu)
 
-    def on_option_update(self, option, value):
-        if option == 'wrap':
-            self.plain_text.set_wrap_mode(value)
-        elif option == 'locked':
-            if value:
-                icon = self.create_icon('lock')
-                tip = _("Unlock")
-            else:
-                icon = self.create_icon('lock_open')
-                tip = _("Lock")
-
-            action = self.get_action(HelpWidgetActions.ToggleLocked)
-            action.setIcon(icon)
-            action.setToolTip(tip)
-        elif option == 'automatic_import':
-            self.object_combo.validate_current_text()
-            self.force_refresh()
-        elif option == 'rich_mode':
-            if value:
-                # Plain Text OFF / Rich text ON
-                self.docstring = not value
-                self.stack_layout.setCurrentWidget(self.rich_text)
-                self.get_action(HelpWidgetActions.ToggleShowSource).setChecked(
-                    False)
-            else:
-                # Plain Text ON / Rich text OFF
-                self.docstring = value
-                self.stack_layout.setCurrentWidget(self.plain_text)
+    @on_conf_change(option='wrap')
+    def on_wrap_option_update(self, value):
+        self.plain_text.set_wrap_mode(value)
 
+    @on_conf_change(option='locked')
+    def on_lock_update(self, value):
+        if value:
+            icon = self.create_icon('lock')
+            tip = _("Unlock")
+        else:
+            icon = self.create_icon('lock_open')
+            tip = _("Lock")
+
+        action = self.get_action(HelpWidgetActions.ToggleLocked)
+        action.setIcon(icon)
+        action.setToolTip(tip)
+
+    @on_conf_change(option='automatic_import')
+    def on_automatic_import_update(self, value):
+        self.object_combo.validate_current_text()
+        if (self._last_editor_doc is None or
+                self._last_console_cb is None or
+                    self._last_editor_cb is None):
+            self.show_intro_message()
```

Comment:
Why is this change necessary? It was not like this in the previous implementation.

I can see that you added these lines in several places below. Perhaps we should move them to a method?

Reply:
This change is necessary in order to display the welcome page in the Help pane due to the propagation of all the configuration options at Spyder startup. If those are not defined, then a `Documentation not available` page would appear instead. The changes do not modify the behaviour of Help at all